app/src/main/java/org/schabi/newpipe/player/VideoPlayerImpl.java

```diff
@@ -889,10 +889,15 @@ private void onMoreOptionsClicked() {
     private void onShareClicked() {
         // share video at the current time (youtube.com/watch?v=ID&t=SECONDS)
         // Timestamp doesn't make sense in a live stream so drop it
-        final String ts = isLive() ? "" : ("&t=" + (getPlaybackSeekBar().getProgress() / 1000));
+
+        String videoUrl = getVideoUrl();
+        final int ts = getPlaybackSeekBar().getProgress() / 1000;
+        if (!isLive() && videoUrl.contains("https://www.youtube.com") && ts >= 0) {
```

Comment:
Can't `ServiceHelper.getSelectedService` be used? It has the advantage of not relying on the specific format of the url.

Reply:
No because selected service isn't always video service. E.g, go to app drawer, select peertube, open a youtube link from an external lihk, and your are playing a youtube video in peertube selected service, and there won't be timestamp even if it should.

Worse, select youtube service (by default when you open the app), open a link except youtube from an external app, there will be timestamp and will break video url

Reply:
Now I check video service id, instead of www.youtube.com (or selected service id)

Reply:
Perfect, I just wanted to suggest this :+1: